backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -379,10 +387,114 @@ const uploadProtectedCompleteHandler = async (
   }
 }
 
+/**
+ * Downloads the file from the vault url and checks that its columns match the attributes provided in the template.
+ * If a template has not yet been uploaded, do not write to the message logs, but prompt the user to upload a template first.
+ * If the template and csv do not match, prompt the user to upload a new file.
+ * @param req
+ * @param res
+ * @param next
+ */
+const tesseractHandler = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<Response | void> => {
+  const { campaignId } = req.params
+  if (await ProtectedService.isProtectedCampaign(+campaignId)) {
+    return res.sendStatus(403)
+  }
+
+  const { url } = req.body
+  const vaultUrl = new URL(url)
+  // check for datasetName is done in isValidVaultUrl middleware
+  const datasetName =
+    vaultUrl.searchParams.get('datasetname') || 'vault dataset'
+  const logMeta = { campaignId, action: 'tesseractHandler' }
+
+  try {
+    // check if template exists
+    const template = await EmailTemplateService.getFilledTemplate(+campaignId)
+    if (template === null) {
+      throw new Error(
+        'Error: No message template found. Please create a message template before uploading a recipient file.'
+      )
+    }
+
+    // Store temp filename
+    await UploadService.storeS3TempFilename(+campaignId, datasetName)
+
+    // Return early because bulk insert is slow
+    res.sendStatus(202)
+
+    try {
+      await retry(async () => {
```

Comment:
```suggestion
      await retry(async (bail) => {
```

If the error is un-retryable (file doesn't exist or it has expired), we should bail to prevent unnecessary retry.